### DISCUSSION TITLE
New pulse once animation for background

### DISCUSSION THREAD

**Discussion by troccoli** [tailwindlabs/tailwindcss#3425] [Help] [answered]
Hello everyone,

What I am trying to do is to create a new animation, `pulse-once`, which will transition from one background color to another and then stop.

I have extended my theme as follows

```js
theme: {
    extend: {
        animation: {
            'pulse-bg-once': 'pulse-bg-once 2s cubic-bezier(0.4, 0, 0.6, 1) forwards'
        },
        keyframes: {
            'pulse-bg-once': {
                from: {opacity: 1},
                to: {opacity: 0}
            }
        }
    },
},
```

and  my HTML is

```html
<div class="animate-pulse-bg-once bg-green-50 hover:bg-gray-100 py-4 flex flex-col md:flex-row">
Content
</div>
```

The background of the parent element is white, so what I want to achive is a transition from a light green to white.

This, obviously, doesn't work as it changes the opacity, so I end up with the whole div being white.

So, I guess I need help in writing the keyframe. I have tried `backgroundOpacity` and `'background-opacity'` but they don't work.

Can anyone help? Is it even possible?

Thanks
Giulio

**Comment by troccoli** [comment, marked as answer]
And I have found the answer.

```js
theme: {
    extend: {
        animation: {
            'pulse-bg-once': 'pulse-bg-once 2s ease-in forwards'
        },
        keyframes: {
            'pulse-bg-once': {
                to: {backgroundColor: 'transparent'}
            }
        }
    },
},
```

This will start from the background color defined in my element, `bg-green-50` in mt case, and will end up with the parent element's background color. I have also changed the function for the transition to `ease-in` as it suits my requirements better.

It would still be nice if there was a way to define the starting and ending background color in the element itself, something like

```html
<div class="animate-pulse-bg-once from-bg-green-200 to-bg-green-50">
```

but I have no idea even where to start with that. If anyone can suggest something please do.

**Comment by CMehlstaeubler** [comment]
Hey @troccoli,

thanks for the code. To extend your solution to have from- to- syntax, I used this in the keyfame:

```        
'pulse-bg': {
    '0%': { backgroundColor: 'var(--tw-gradient-from)' },
    '50%': { backgroundColor: 'var(--tw-gradient-to)' },
    '100%': { backgroundColor: 'var(--tw-gradient-from)' },
},
```

The syntax on the html element would be ` <div class="animate-pulse-bg from-emotion-500 to-emotion-600">`

(I changed it to bounce between two colours, but the principle should be the same for a single pulse.)

Hope this helps someone in the future.
